activesupport/test/cache/stores/mem_cache_store_test.rb

```diff
@@ -34,6 +46,76 @@ def setup
   include EncodedKeyCacheBehavior
   include AutoloadingCacheBehavior
 
+  def test_connection_pool
+    emulating_latency do
+      begin
+        mutex = Mutex.new
+        cache = ActiveSupport::Cache.lookup_store(:mem_cache_store, pool_size: 1, pool_timeout: 1)
```

Comment:
Seems like it's probably worth testing that the pool actually works, too, not just that it applies a timeout: let's also run this with `pool_size: 2`, and prove the second request finishes while the slow thread is still stuck.